hnsecure/src/test/java/ca/bc/gov/hlth/hnsecure/route/RouteTest.java

```diff
@@ -84,6 +85,34 @@ public void testSuccessfulJMBMessage() throws Exception {
 
 		context.stop();
 	}
+
+	@Test
+	public void testCustomTransactionIdFromHttpHeader() throws Exception {
+		context.start();
+
+		// Set expectations
+		getMockEndpoint("mock:response").expectedMessageCount(1);
+		responseEndpoint.expectedBodiesReceived(WRAPPED_R03_RESPONSE);
+		// Not an ideal way to test if the code to set the exchangeId works, but the mock ResponseEndpoint doesn't
+		// return the updated ExchangeId even though it shows correctly in all logging.
+		// i.e. assertEquals("test-request-id", responseEndpoint.getReceivedExchanges().get(0).getExchangeId()) fails
+		// even though the code works
+		// Instead just test if the code is called
+		getMockEndpoint("mock:SetExchangeIdFromHeader").expectedMessageCount(1);
```

Comment:
Comment explains it. Couldn't find a way to test the actual ExchangeId so instead I'm just testing if the processor is called.

Reply:
Out of curiousity I had a look at this as well. There's a lot of copying of Exchanges behind the scenes in the mock classes. It looks like the copy that you end up getting in the responseEndpoint which is used in the assertion gets created before SetExchangeIdFromHeader is invoked and hence it has the original, default exchangeId. So, it seems like an artifact of testing and not worth putting any more time into resolving.